File: NumberofSegmentsinaString2.py
class Solution:
    def countSegments(self, s):
        """
        :type s: str
        :rtype: int
        """
        count = 0
        for i in range(len(s)):
            if s[i] != ' ' and (i == len(s) - 1 or s[i + 1] == ' '):
                count += 1
        return count


if __name__ == '__main__':
    s = input()
    solution = Solution()
    print(solution.countSegments(s))

File: NumberofSegmentsinaString2.cpp
#include<iostream>
#include<vector>

using namespace std;

class Solution {
public:
    int countSegments(string s) {
        int count = 0;
        
        for(int i = 0; i < s.size(); i++) {
        	if((i == 0 || s[i - 1] == ' ') && s[i] != ' ') count++;
		}
		
		return count;
        
    }
};

int main() {
	
	string s;
	getline(cin, s);
	
	Solution *solution = new Solution();
	cout<<solution->countSegments(s);
	
	return 0;
	
} 
